volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+            for net_dev in net.dev_base_head.to_list(net_device_symname, "dev_list"):
+                if net.get_inode() != netns_id:
+                    continue
+                dev_name = utility.array_to_string(net_dev.name)
+                netdevices_map[net_dev.ifindex] = dev_name
+        return netdevices_map
+
+    def process_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str], Dict]:
+        """Takes a kernel generic `sock` object and processes it with its respective socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns a tuple with:
+            sock: The respective kernel's *_sock object for that socket family
+            sock_stat: A tuple with the source, destination and state strings.
+            extended: A dictionary with key/value extended information.
+        """
+        family = sock.get_family()
+        extended = {}
+        sock_handler = self._sock_family_handlers.get(family)
+        if sock_handler:
+            try:
+                unix_sock, sock_stat = sock_handler(sock)
+                self._update_extended_socket_filters_info(sock, extended)
+
+                return unix_sock, sock_stat, extended
+            except exceptions.SymbolError as e:
+                # Cannot finds the *_sock type in the symbols
+                vollog.log(constants.LOGLEVEL_V, "Error processing socket family '%s': %s", family, e)
+        else:
+            vollog.log(constants.LOGLEVEL_V, "Unsupported family '%s'", family)
+
+        # Even if the sock family is not supported, or the required types
+        # are not present in the symbols, we can still show some general
+        # information about the socket that may be helpful.
+        saddr_tag = daddr_tag = "?"
+        state = sock.get_state()
+
+        sock_stat = saddr_tag, daddr_tag, state
+
+        return sock, sock_stat, extended
+
+    def _update_extended_socket_filters_info(self, sock: objects.Pointer, extended: dict) -> None:
+        """Get infomation from the socket and reuseport filters
+
+        Args:
+            sock: The kernel sock (sk) struct
+            extended: Dictionary to store extended information
+        """
+        if sock.has_member("sk_filter") and sock.sk_filter:
+            sock_filter = sock.sk_filter
+            extended["filter_type"] = "socket_filter"
+            self._extract_socket_filter_info(sock_filter, extended)
+
+        if sock.has_member("sk_reuseport_cb") and sock.sk_reuseport_cb:
+            sock_reuseport_cb = sock.sk_reuseport_cb
+            extended["filter_type"] = "reuseport_filter"
+            self._extract_socket_filter_info(sock_reuseport_cb, extended)
+
+    def _extract_socket_filter_info(self, sock_filter: objects.Pointer, extended: dict) -> None:
+        extended["bpf_filter_type"] = "cBPF"
+
+        if not sock_filter.has_member("prog") or not sock_filter.prog:
+            return
```

Comment:
Similar to continue above, it looks like everything following could simply be under the inverse of this guard statement?  This is definitely a stylistic thing, so not a show stopper, but as I say I find it interrupts the flow when reading back the code.